File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/InsertNoteAsyncTask.java
package com.example.user.biblejournal.model.asynctasks;

import android.os.AsyncTask;

import com.example.user.biblejournal.model.database.note.NoteDao;
import com.example.user.biblejournal.model.database.note.NoteEntity;

public class InsertNoteAsyncTask extends AsyncTask<NoteEntity, Void, Void> {
    final NoteDao noteDao;

    public InsertNoteAsyncTask(NoteDao noteDao) {
        this.noteDao = noteDao;
    }

    @Override
    protected Void doInBackground(NoteEntity... noteEntities) {
        noteDao.insertNote(noteEntities[0]);
        return null;
    }
}

File: app/src/main/java/com/example/user/biblejournal/model/asynctasks/InsertNoteAsyncTask.kt
package com.example.user.biblejournal.model.asynctasks

import android.os.AsyncTask

import com.example.user.biblejournal.model.database.note.NoteDao
import com.example.user.biblejournal.model.database.note.NoteEntity

class InsertNoteAsyncTask(internal val noteDao: NoteDao) : AsyncTask<NoteEntity, Void, Void>() {

    override fun doInBackground(vararg noteEntities: NoteEntity): Void? {
        noteDao.insertNote(noteEntities[0])
        return null
    }
}